testcontainers/core/docker_client.py

```diff
@@ -30,15 +32,17 @@ def run(self, image: str,
             stdout: bool = True,
             stderr: bool = False,
             remove: bool = False, **kwargs) -> Container:
-        return self.client.containers.run(image,
-                                          command=command,
-                                          stdout=stdout,
-                                          stderr=stderr,
-                                          remove=remove,
-                                          detach=detach,
-                                          environment=environment,
-                                          ports=ports,
-                                          **kwargs)
+        return ResourceCleaner.instance().attach(
```

Comment:
Thoughts on just registering the hook directly here rather than going via the `ResourceCleaner`? E.g. `atexit.register(container.stop)` (with some additional error checks).

Reply:
Yes, thanks for your advice. Now code look more simpler. 

I combined `stop_silent` and `atexit.register(stop_silent(container))` because some times container does not exists in the end so we have to suppress `docker.errors.NotFound`. 

```python
def stop_silent(container):
    def wrapper():
        try:
            container.stop()
        except NotFound:
            pass
        except Exception as e:
            logger.exception(e)
    return wrapper
```

Reply:
Nice! `atexit.register` even lets you pass in arguments. So you could further simplify to

```python
def stop_silent(container):
    try:
        container.stop()
    except NotFound:
        pass
    except Exception as e:
        logger.exception(e)
```

and then follow up with

```python
atexit.register(stop_silent, container)
```

Reply:
Thanks, I applied your fix.